### ISSUE TITLE
rclone WebDAV doesnt allow to login to the Sharepoint Online: AADSTS50053: The account is locked, you've tried to sign in too many times with an incorrect user ID or password.

### ISSUE THREAD

**Issue by Nargizzz** [rclone/rclone#4400]
I have a Sharepoint Online site, which has a sub-site with restricted access.
To get there I chose to use webdav.

Example URLs: 
The site URL: https://orgname.sharepoint.com/sites/siteID/Shared%20Documents
Sub-site URL: https://orgname.sharepoint.com/sites/siteID/Restricted/Shared%20Documents

So I was not able to reach sub-site with "onedrive" and used webdav instead. But it seems webdav is looking for some kind of cookies or smth like this, because it is not allowing me to login with my credentials and now even blocked account.

Will be happy if you could help with this!

**Comment by ncw** [comment]
> So I was not able to reach sub-site with "onedrive"

What did you try?

Did you try a graph search?



> But it seems webdav is looking for some kind of cookies or smth like this, because it is not allowing me to login with my credentials and now even blocked account.

Did you make a sharepoint WebDAV backend? That should get the right cookies for you.

**Comment by Nargizzz** [comment]
I am searching to find necessary site via "search" and then it it prompts me to choose a drive I want to use. However, the list of these drives are under "Shared Documents" (documentlibrary) folder only and obviously don't contain the drive I need.
Site URL: https://orgname.sharepoint.com/sites/siteID
The drive URL: https://orgname.sharepoint.com/sites/siteID/Shared%20Documents/SOME_FILES
Sub-site URL: https://orgname.sharepoint.com/sites/siteID/Restricted/Shared%20Documents

So I decided to use WebDAV, but it doesn't work for me because when I use only username and password it says that they are wrong. My sharepoint is Federated. I have found, that using WebDAV with cookies can solve the issue, but I am not sure.

Could you please suggest how to make sharepoint WebDAV backend?

**Comment by ncw** [comment]
Did you answer this question like this?

```
Choose a number from below, or type in your own value
 1 / Nextcloud
   \ "nextcloud"
 2 / Owncloud
   \ "owncloud"
 3 / Sharepoint
   \ "sharepoint"
 4 / Other site/service or software
   \ "other"
vendor> 3       
```

That should be enough to get the cookie and work with sharepoint.

**Comment by Nargizzz** [comment]
Yes! I chose "sharepoint" oprion.

I got a reply from organization IT Support that we use ADFS+MFA authentication. Meaning that I have to login only through web browser.

Is there a way in webdav option to use web browser authentication as it is in onedrive?
Because it seems onedrive isn't able to list subsites in a drive list.

**Comment by ncw** [comment]
If you could get the cookies from your browser, then it would be possible to put them in rclone.

However I think the cookies might only last for a limited time 12 hours?

Do you want to try doing this?

The relevant code is here

https://github.com/rclone/rclone/blob/4f7f5404cea5211c69114ffcd79931edf44b6b99/backend/webdav/webdav.go#L442-L468

We'd need a way of storing the cookings in the config file. You could try just pasting them into that bit of code to start with to see if the idea will work at all.

**Comment by animosity22** [comment]
Anything to do here?

**Comment by ncw** [comment]
Being able to store or add cookies for the webdav config would be a nice idea, but I don't have time to work on it at the moment.

This particular problem is something to do with ADFS+MFA authentication which I know very little about.

@Nargizzz is this still a problem?

**Comment by Nargizzz** [comment]
The issue is resolved for me. This was a task at my company, where we use ADFS authentication. I will share how we solved this task as I saw a lot of people have the same problem.
In the Azure portal administrator added service principal with the subscription ID, which I later attached to the sharepoint site.
I used Office365-REST-Python-Client open source project to upload files.

If I am not mistaken, I could do the same with rclone. Just azure admin had to "allow" rclone application in the portal.
